com.unity.render-pipelines.universal/Editor/AssetPostProcessors/MaterialPostprocessor.cs

```diff
@@ -154,13 +161,23 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
                     {
                         assetVersion.version = k_Upgraders.Length;
                         s_CreatedAssets.Remove(asset);
-                        InitializeLatest(material, id);
+                        InitializeLatest(material, shaderID);
                         debug += " initialized.";
                     }
                     else
                     {
-                        assetVersion.version = UniversalProjectSettings.materialVersionForUpgrade;
-                        debug += $" assumed to be version {UniversalProjectSettings.materialVersionForUpgrade} due to missing version.";
+                        if (shaderID.IsShaderGraph())
+                        {
+                            // ShaderGraph materials NEVER had asset versioning applied prior to version 5.
+                            // so if we see a ShaderGraph material with no assetVersion, set it to 4 to ensure we apply all necessary versions.
+                            assetVersion.version = 5;
+                            debug += $" shadergraph material assumed to be version 4 due to missing version.";
```

Comment:
Should it be 4 or 5?

Reply:
Why not assume `UniversalProjectSettings.materialVersionForUpgrade`?

Reply:
The number should be the version at which ShaderGraph Material Upgrading was first introduced.  Prior to that version, the Material Upgrader ignored ShaderGraph based Materials entirely.   Without doing this, if we assume latest version, it will not run the upgrader on any pre-existing ShaderGraph Materials, assuming they have already been upgraded.  This was a problem when I had an upgrade pass to apply keywords to all pre-existing ShaderGraph Materials.  That pass was removed as unnecessary now that we set "Allow Material Override" to false by default -- since no keywords are needed in that mode (it's effectively the same as the old behavior), it no longer needs to update any pre-existing graphs.   I left it in there though in case we do add upgraders that affect ShaderGraph in the future, as we can't assume latest version for ShaderGraphs.

Reply:
It was 4, but someone added another upgrader pass which got merged in, so I bumped it to 5.  I will fix the comment to match.